src/DebugBar/Resources/debugbar.js

```diff
@@ -1293,4 +1293,88 @@ if (typeof(PhpDebugBar) == 'undefined') {
 
     });
 
+    // ============= Handling Dragging Behavior ============
+    var storedPosX = localStorage.getItem('phpdebugbarPositionX') || 0;
+    var screenWidth = $(window).width();
+
+    $(document).on('click', '.phpdebugbar', function () {
+        let isClosed = $('.phpdebugbar').hasClass('phpdebugbar-closed');
+        if (!isClosed) {
+            // Save current position to localStorage when closing the debugbar
+            storedPosX = localStorage.getItem('phpdebugbarPositionX');
+            $('.phpdebugbar').css('left', '0px');
+        } else {
+            // Restore saved position from localStorage when opening the debugbar
+            localStorage.setItem('phpdebugbarPositionX', storedPosX);
+            $('.phpdebugbar').css('left', storedPosX + 'px');
+        }
+    });
+
+    $(document).ready(function () {
```

Comment:
instead of waiting for jquery to load, it should be executed in the debugbar render